flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/IOManager.java

```diff
@@ -54,6 +56,14 @@
     protected IOManager(String[] tempDirs) {
         this.fileChannelManager =
                 new FileChannelManagerImpl(Preconditions.checkNotNull(tempDirs), DIR_NAME_PREFIX);
+        if (LOG.isInfoEnabled()) {
+            LOG.info(
+                    "Created a new {} for spilling of task related data to disk (joins, sorting, ...). Used directories: {}.",
+                    FileChannelManager.class.getSimpleName(),
+                    Arrays.stream(fileChannelManager.getPaths())
+                            .map(File::getAbsolutePath)
+                            .collect(Collectors.joining(", ")));
```

Comment:
maybe do something fancy here because the directories might be quite long:
```
Used Directories:
     <directory1>
     <directory2>
```

Reply:
I'd strongly recommend against multi-line log messages, because this would affect logging infrastructure of many users.

For tools such as fluentd, filebeat, logstash, that rely on log parsing, this would require special handling [1]. One special case of multiline log are stacktraces, but these have a well defined format and are usually supported out of the box.

[1] https://www.elastic.co/guide/en/beats/filebeat/current/multiline-examples.html#multiline-examples

Reply:
We already have a various logging statements that use multi-line log messages without any user ringing alarm bells, so I'm not too worried at this point.